server/session/security_cookie.ts

```diff
@@ -59,7 +59,7 @@ export function getSecurityCookieOptions(
       // TODO: with setting redirect attributes to support OIDC and SAML,
       //       we need to do additional cookie validation in AuthenticationHandlers.
       // if SAML fields present
-      if (sessionStorage.saml && sessionStorage.saml.requestId && sessionStorage.saml.nextUrl) {
+      if (sessionStorage.saml && sessionStorage.saml.requestId) {
         return { isValid: true, path: '/' };
```

Comment:
is the expected behavior to return '/' regardless of nextUrl being present?

Reply:
The only usage I could find of `path` is [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/http/cookie_session_storage.ts#L165). The [Path Attribute](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#path_attribute) in this case means the root of the opensearch-dashboards application. 

I need to look into this more, because the [cookie_session_storage.ts](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/http/cookie_session_storage.ts#L157) from core does take the basePath into account.

Reply:
So this is and isn't a bug at the same time. Technically, this should be the basePath as you pointed out, but its not a bug because there are multiple places that can clear cookies.

If the cookie is not valid, there is logic in OSD core [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/http/cookie_session_storage.ts#L164-L166) to clear the invalid cookie corresponding to the same origin and path. Because of this bug, if a dashboards instance has a basePath, that function will never be called.

That being said, the security-dashboards-plugin will also [clear all cookies](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/authentication_type.ts#L148) for dashboards if the cookie is invalid so they will still be cleared.